### ISSUE TITLE
please fix template for v6

### ISSUE THREAD

**Issue by billnbell** [logstash-plugins/logstash-output-elasticsearch#734]
Please fix v6.

https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/lib/logstash/outputs/elasticsearch/elasticsearch-template-es6x.json

"template" : "logstash-*", -- deprecated
"_default_" : { -- deprecated

Should be index-pattern, and _doc

**Comment by jakelandis** [comment]
@billnbell - `_doc`  was not available until the most recent `6.2.0` release [via](https://github.com/elastic/elasticsearch/pull/27816).  I am not sure if we want to introduce a 6.2+ only change to the matrix. This [has already been](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/lib/logstash/outputs/elasticsearch/elasticsearch-template-es7x.json) changed to `_doc` for 7.0.0.

template vs. index-pattern - We missed that [memo](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/breaking_60_indices_changes.html#_index_templates_use_literal_index_patterns_literal_instead_of_literal_template_literal) for 6.2.  We will correct for 7.0 certainly.  Again not sure if we want a 6.2+ only change in our matrix. 

@andrewvc / @jordansissel - thoughts on adopting these changes for 6.x ?

**Comment by wkruse** [comment]
Related to #735, #738.

**Comment by jsvd** [comment]
there's currently no feature in this plugin to conditionally build parts of the templated based on the elasticsearch cluster version. the major version of the cluster is computed and the corresponding template is used.
In order to prevent breaking compatibility, I'd avoid performing any changes for 6.x and introduce the changes for 7.x, unless there are some arguments against this.

**Comment by jakelandis** [comment]
> I'd avoid performing any changes for 6.x 

I agree, else we get one (or more) things to watch out for in the compatibility matrix.  If we did make this change, we would need to release a 6.1 version Logstash with the ES output version pinned. 

With 6.2+ I believe the current defaults result in a warning in the log and a bit of mis-match with the documentation, but no technical issues.  If we changed this, we could be introducing a technical issue for versions prior to 6.2.

We should however, update 7.x  `template -> index-pattern`
